### PR TITLE
Perform switch to CMSMonitoring/StompAMQ

### PR DESCRIPTION
**Pull request by vkuznet**
This PR contains changes to perform switch to dmwm/CMSMonitoring StompAMQ version.

@amaltaro , Alan please advise what other changes will be required to properly setup/build WMCore components. I already adjusted requirements.txt and setup scripts.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -21,5 +21,6 @@ retry==0.9.1
 setuptools==39.2.0
 stomp.py==4.1.15
 rucio-clients==1.19.3
+CMSMonitoring>=0.1.1
```
Comment (amaltaro) [inline requirements.txt]: From another thread, I've seen this version is broken. But don't bother about updating it, there might be other things to be fixed when pushing it in and then I'll just create a new PR.

**Comment by vkuznet** [comment]
I updated the CMSMonitoring version in requirements.txt
Please review and merge.

**Comment by amaltaro** [comment]
For the record, a fix and a new CMSMonitoring release need to be made, see issue:
https://github.com/dmwm/CMSMonitoring/issues/28

**Comment by vkuznet** [comment]
Alan, please update `CMSMonitoring>=0.1.9` to `CMSMonitoring>=0.2.2` (the new release should appear on pypi today, there is usually some delay in new release appearnce over there)

**Comment by vkuznet** [comment]
ohh, that my PR (I need to wake up), so I'll update it.

**Comment by amaltaro** [comment]
Thanks! We're still relying on the RPM for the WMCore services though.
BTW, can you please squash these commits. I believe changes are fine, but I still didn't manage to test it thoroughly.

**Comment by vkuznet** [comment]
Done.

**Comment by vkuznet** [comment]
More info on running CMS Monitoring version of StompAMQ:
- new code resolve CERN broker names into series of IP address (both ipv4 and ipv6), if resolution is successful it uses all of them, therefore the node where this version should run should provide ipv6 capability, e.g. ping6 command should work
- the connection to broker are picked up randomly to provide access balancing approach to brokers
- the code uses internal connection pool
- users of the code can either provide username/password or key/cert

**Comment by amaltaro** [comment]
Thanks for the additional information, Valentin. My VM seems to be ipv6 enabled and I can `ping6` those ipv6 address that get resolved (`ip -6 a` also returns a v6 address).

Unfortunately, I still see problems connecting to the AMQ brokers and it takes ~10min to setup those connections, and it fails a couple of times starting the connection, this line:
https://github.com/dmwm/CMSMonitoring/blob/master/src/python/CMSMonitoring/StompAMQ.py#L214

Error message and traceback isn't that helpful either (host information removed, and additional worthless logging added):
```
2019-07-03 22:07:10,967:INFO:StompAMQ:host and ports: [('cms-mb.cern.ch', XXX)]
2019-07-03 22:07:10,968:INFO:StompAMQ:resolver: [(XXX)]
2019-07-03 22:13:33,546:ERROR:StompAMQ:Jul 03 2019 20:13:33, connection to [('cms-mb.cern.ch', XXX)] failed. Error: 
Available hosts: [XXXX]args: (), message: 
Traceback (most recent call last):
  File "/data/srv/HG1907f-comp2/sw.amaltaro/slc7_amd64_gcc630/external/cmsmonitoring/0.2.1/lib/python2.7/site-packages/CMSMonitoring/StompAMQ.py", line 214, in connect
    conn.start()
  File "build/bdist.linux-x86_64/egg/stomp/connect.py", line 46, in start
    self.transport.start()
  File "build/bdist.linux-x86_64/egg/stomp/transport.py", line 107, in start
    self.attempt_connection()
  File "build/bdist.linux-x86_64/egg/stomp/transport.py", line 767, in attempt_connection
    raise exception.ConnectFailedException()
ConnectFailedException
2019-07-03 22:19:56,266:ERROR:StompAMQ:Jul 03 2019 20:19:56, connection to [('cms-mb.cern.ch', XXX)] failed. Error: 
...
```

**Comment by vkuznet** [comment]
Alan, do you use certificates or login/pwd authentication schema.
While performing tests with certificates I also see connection failures to
brokers but I can connect fine with my login/pwd credentials.

Please provide name/IPs of nodes you're trying to connect from since
I'm in discussion with Lionel from CERN IT about these failures.
So I can pass around this info to him.

FYI, in https://github.com/dmwm/CMSMonitoring/pull/30 I added also
ipv4_only option to avoid issues with ipv6 connections. I'm awaiting
a word from CERN IT before merging the my PR too.

Valentin.

**Comment by amaltaro** [comment]
I've just sent you a private email with further info.

**Comment by vkuznet** [comment]
Alan, per our discussion on a meeting I changed version of CMSMonitoring/StompAMQ to 0.2.4 which now has `ipv4_only=True` as default parameter in constructor of StompAMQ. This will avoid issues with ipv6 connection failures.

Also, since you mentioned, you need to adjust WMCore code to properly capture parameters from `make_notification` API. Here is an example how code should be used:
```
amq = StompAMQ(...) # get new instance of StompAMQ

doc = {} # some document
doc_type = "bla" # document type
# create notification objects
notification, okeys, ukeys = amq.make_notification(doc, doc_type)
```
The `okeys` and `ukeys` list of offending and unknown keys, respectively. These are used by validation code and will report to client what does not conform the schema. For time being if you do not plan to use validation you can simply ignore them, so you can do
```
notification, _, _ = amq.make_notification(doc, doc_type)
```

**Comment by amaltaro** [comment]
Thanks, Valentin. I'll give it another try tomorrow and if everything goes well, build a new CMSMonitoring RPM.

**Comment by amaltaro** [comment]
Thanks, Valentin. I cherry-picked your changes and create a new PR with further needed changes:
https://github.com/dmwm/WMCore/pull/9292

**Pull request closed without merging**